app/cdap/components/DirectiveInput/index.tsx

```diff
@@ -0,0 +1,269 @@
+  position: fixed;
+  bottom: 93px;
+  width: 100%;
+`;
+
+const CloseIconButton = styled(IconButton)`
+  color: #ffffff;
+`;
+
+export default function({
+  columnNamesList,
+  onDirectiveInputHandler,
+  onClose,
+}: IDirectiveInputProps) {
+  const [directiveInput, dispatch] = useReducer(reducer, initialDirectiveInputState);
+  const {
+    inputDirective,
+    isDirectiveSet,
+    directivesList,
+    appliedDirective,
+    directiveColumnCount,
+    isDirectivePaste,
+    directiveUsageList,
+    directiveColumns,
+    enterCount,
+  } = directiveInput;
+  const directiveRef = useRef();
+
+  const handleDirectiveChange = (value) => {
+    if (!value) {
+      dispatch({
+        type: IDirectiveActions.SET_NO_DIRECTIVE,
+        payload: {
+          inputDirective: false,
+          enterCount: 0,
+        },
+      });
+    }
+    const inputText = value.split(' ');
+    const firstIndexInputTextValue = inputText[0];
+    if (directivesList.findIndex((el) => el.directive == firstIndexInputTextValue) !== -1) {
+      dispatch({
+        type: IDirectiveActions.DIRECTIVES_APPLIED_SET_COUNT,
+        payload: {
+          directiveColumnCount: getDirectiveColumnCount(firstIndexInputTextValue),
+          isDirectiveSet: true,
+          appliedDirective: [firstIndexInputTextValue],
+        },
+      });
+    }
+    if (isDirectiveSet) {
+      columnNamesList.forEach((column) => {
+        if (value.indexOf(column.label) !== -1) {
+          dispatch({
+            type: IDirectiveActions.DIRECTIVES_COLUMNS_LIST,
+            payload: {
+              directiveColumns: !directiveColumns.includes(column.label)
+                ? [...directiveColumns, column.label]
+                : directiveColumns,
+              enterCount: appliedDirective.length + directiveColumns.length,
+            },
+          });
+        }
+      });
+    }
+    dispatch({
+      type: IDirectiveActions.INPUT_DIRECTIVE,
+      payload: value,
+    });
+  };
+
+  const getDirectiveColumnCount = (firstIndexInputTextValue) => {
+    if (TWO_COLUMN_DIRECTIVE.includes(firstIndexInputTextValue)) {
+      return 2;
+    } else if (MULTIPLE_COLUMN_DIRECTIVE.includes(firstIndexInputTextValue)) {
+      return 0;
+    } else {
+      return 1;
+    }
+  };
+
+  useEffect(() => {
+    dispatch({
+      type: IDirectiveActions.DIRECTIVE_USAGE_LIST,
+      payload: formatDirectiveUsageData(inputDirective, directivesList),
+    });
+  }, [inputDirective]);
+
+  const handleKeyDownEvent = (KeyboardEvent: React.KeyboardEvent<HTMLInputElement>) => {
+    if ((KeyboardEvent.ctrlKey || KeyboardEvent.metaKey) && KeyboardEvent.keyCode == 86) {
+      dispatch({
+        type: IDirectiveActions.DIRECTIVE_PASTE,
+        payload: true,
+      });
+    } else if (KeyboardEvent.key === 'Enter') {
+      dispatch({
+        type: IDirectiveActions.SET_ENTER_COUNT,
+        payload: onDirectiveComplete(),
+      });
+    }
+  };
+
+  const onDirectiveComplete = () => {
+    if (
```

Comment:
nit: these `if` statements all do the same thing, you can chain the conditions by `||` and use only one block.

Reply:
Done